crates/ika-move-packages/packages/ika_system/sources/dwallet/dwallet_2pc_mpc_coordinator_inner.move

```diff
@@ -1766,51 +1901,56 @@ public(package) fun is_ecdsa_presign_valid(
 /// # Aborts
 /// - **`EExtraDataAndMessagesLenMismatch`**: If `signature_algorithm_data` and `message_approvals` have different lengths.
 /// - **`EMissingApprovalOrWrongApprovalOrder`**: If message approvals are incorrect or missing.
-fun emit_ecdsa_sign_event(
+fun emit_sign_event(
     self: &mut DWalletCoordinatorInner,
     pricing: PricingPerOperation,
     payment_ika: &mut Coin<IKA>,
     payment_sui: &mut Coin<SUI>,
     message_approval: MessageApproval,
-    presign_cap: ECDSAPresignCap,
+    presign_cap: PresignCap,
     message_centralized_signature: vector<u8>,
     is_future_sign: bool,
     ctx: &mut TxContext
 ) {
     let created_at_epoch = self.current_epoch;
-    let (dwallet, public_output) = self.get_active_dwallet_and_public_output_mut(presign_cap.dwallet_id);
 
-    assert!(dwallet.ecdsa_presigns.contains(presign_cap.presign_id), EPresignNotExist);
-    let presign = dwallet.ecdsa_presigns.remove(presign_cap.presign_id);
+    assert!(self.presigns.contains(presign_cap.presign_id), EPresignNotExist);
+    let presign = self.presigns.remove(presign_cap.presign_id);
```

Comment:
We need to be very careful here, if you remember we had some issues in the previous design about presigns removals etc. and the structure is changed now, and I think some of the reasons that the issues weren't serious last time was because the presigns were kept in the dwallet itself.